bson/src/main/org/bson/codecs/configuration/mapper/conventions/BeanPropertiesConvention.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2008-2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson.codecs.configuration.mapper.conventions;
+
+import org.bson.codecs.configuration.mapper.ClassModel;
+import org.bson.codecs.configuration.mapper.FieldModel;
+import org.bson.codecs.configuration.mapper.MethodModel;
+import org.bson.codecs.configuration.mapper.Weights;
+
+import java.util.List;
+
+/**
+ * This convention ensures only fields that are private, nonstatic, nontransient, and nonfinal that have a proper getter and setter as per
+ * the Beans Specification are included in BSON processing.
+ *
+ * @see <a target="_blank" href="http://www.oracle.com/technetwork/articles/javaee/spec-136004.html">JavaBeans Spec</a>
+ */
+public class BeanPropertiesConvention implements Convention {
+    private Integer getWeight() {
+        return Weights.BUILT_IN_CONVENTION;
```

Comment:
Since this is the default convention for mapping, this should probably be DEFAULT.